lib/lambda_ethereum_consensus/state_transition/epoch_processing.ex

```diff
@@ -84,6 +84,43 @@ defmodule LambdaEthereumConsensus.StateTransition.EpochProcessing do
     {:ok, new_state}
   end
 
+  @spec process_slashings(BeaconState.t()) :: {:ok, BeaconState.t()}
+  def process_slashings(%BeaconState{validators: validators, slashings: slashings} = state) do
+    epoch = Accessors.get_current_epoch(state)
+    total_balance = Accessors.get_total_active_balance(state)
+
+    proportional_slashing_multiplier = ChainSpec.get("PROPORTIONAL_SLASHING_MULTIPLIER_BELLATRIX")
+    epochs_per_slashings_vector = ChainSpec.get("EPOCHS_PER_SLASHINGS_VECTOR")
+    effective_balance_increment = ChainSpec.get("EFFECTIVE_BALANCE_INCREMENT")
+
+    slashed_sum = Enum.reduce(slashings, 0, &+/2)
+
+    adjusted_total_slashing_balance =
+      min(slashed_sum * proportional_slashing_multiplier, total_balance)
+
+    increment = effective_balance_increment
+
+    validators_indices = Enum.with_index(validators)
+
+    new_state =
+      Enum.reduce(validators_indices, state, fn {validator, index}, acc ->
```

Comment:
Nit: we should use pipes to avoid temporary variables
```suggestion
    new_state =
      validators
      |> Enum.with_index()
      |> Enum.reduce(state, fn {validator, index}, acc ->
```